### DISCUSSION TITLE
Including properties in event stream

### DISCUSSION THREAD

**Discussion by Toudahl** [datalust/seq-tickets#1937] [Help] [answered]
Hello,

I am trying to set up slack notifications for an event stream, where i want to include specific properties.
I have successfully set up the connection, and the events matching the filter i have configured arrives in my slack channel.

Now i am trying to include the properties i want into the message. However this is causing some problems.

I would like to include the log category, and the application name.
The log category shows up as expected, but i cannot make the application name show up.

I cannot seem to find the correct documentation for this. Perhaps it is because im too new with seq, so i dont know the vocabulary yet.
Can someone direct me to the documentation that i have missed, and perhaps even give me the answer to this?

So far I have tried the following guesses:
`ApplicationName,dotnet.ilogger.category`
`ra.ApplicationName,dotnet.ilogger.category`
`$ra.ApplicationName,dotnet.ilogger.category`
`@ra.ApplicationName,dotnet.ilogger.category`
`@Resource.ApplicationName,dotnet.ilogger.category`
`service.name,dotnet.ilogger.category`
`@Resource['service.name'],dotnet.ilogger.category`

This is an example log message:
```json
{
    "@i": "9615c48b",
    "@l": "Warning",
    "@m": "Suppressed Exception",
    "@mt": "Suppressed Exception",
    "@ra": {
        "ApplicationName": "ddns-service",
        "EnvironmentName": "Development",
        "service.instance.id": "69a17202-9448-4622-b369-d8a304682b2b",
        "service.name": "ddns-service",
        "telemetry.sdk.language": "dotnet",
        "telemetry.sdk.name": "opentelemetry",
        "telemetry.sdk.version": "1.5.1"
    },
    "@t": "2023-09-08T12:12:32.3208300Z",
    "@x": "System.Exception: test streaming\r\n   at DDNS_Updater.Services.DdnsHostedService.ExecuteAsync(CancellationToken stoppingToken) in D:\\code\\DDNS-Updater\\DDNS-Updater\\Services\\DDNSHostedService.cs:line 44",
    "ClientTimestamp": "2023-09-08T12:12:31.3040665Z",
    "dotnet.ilogger.category": "DDNS_Updater.Services.DdnsHostedService",
    "Id": 13,
    "Name": "SuppressedException"
}
```

**Comment by nblumhardt** [comment, marked as answer]
Hi @Toudahl - thanks for dropping us a line!

At present the Slack app only supports root-level property expressions, so to get the application name in you need to pull in `@ra` - all of the resource attributes, unfortunately.

I've just been doing some work in this space for the email apps over at https://github.com/datalust/seq-app-mail - the templating system there is designed to deal with these newer/more complex cases. I'll start a discussion on https://github.com/bytenik/Seq.App.Slack about whether we could make use of this over there, too 👍

**Reply by nblumhardt** [reply]
Raised https://github.com/bytenik/Seq.App.Slack/issues/68

**Comment by Toudahl** [comment]
Thanks. I hope this will be supported :-)

One minor detail, the attribute "ApplicationName" is a custom one I made. The semantically correct one is "service.name".

I don't know if it would be relevant to update your example in the linked issue.
Regardless, I got my answer - so I'm closing this here.
Have a nice day

**Reply by nblumhardt** [reply]
Thanks for the follow-up 👍
